File: patterns/C++/pattern15.cpp
/*
Generate the following pattern
* * * * * * * * * *
* * * *     * * * *
* * *         * * *
* *             * *
*                 *
*                 *
* *             * *
* * *         * * *
* * * *     * * * *
* * * * * * * * * *
*/

#include<bits/stdc++.h>
using namespace std;

void pattern15(int n){
    for(int i=0; i<n; i++){
        for(int j=0; j<(n-i); j++){
            cout << "*" <<" ";
        }

        for(int j=0; j<4*i; j++){
            cout << " ";
        }

        for(int j=0; j<(n-i); j++){
            cout << "*" << " ";
        }

        cout << endl;
    }

    for(int i=1; i<=n; i++){
        for(int j=1; j<=i; j++){
            cout << "*" <<" ";
        }

        for(int j=1; j<=4*(n-i); j++){
            cout << " ";
        }

        for(int j=i; j>0; j--){
            cout << "*" << " ";
        }

        cout << endl;
    }

}


int main(){
    int n;
    cout << "Enter the Number of rows: ";
    cin >> n;

    cout << "\nHere is the generated pattern\n";
    pattern15(n);
}

File: patterns/Python/pattern15.py
"""
Generate the following pattern
* * * * * * * * * *
* * * *     * * * *
* * *         * * *
* *             * *
*                 *
*                 *
* *             * *
* * *         * * *
* * * *     * * * *
* * * * * * * * * *
"""

def pattern15(n):
    for i in range(1, n+1):
        for j in range(n+1-i, 0, -1):
            print("*", end=" ")
            
        for j in range(4*(i-1)):
            print(" ", end="")
            
        for j in range(n+1-i, 0, -1):
            print("*", end=" ")
                        
        print("")
        
    for i in range(1, n+1):
        for j in range(1, i+1):
            print("*", end=" ")
            
        for j in range(4*(n-i)):
            print(" ", end="")
            
        for j in range(i, 0, -1):
            print("*", end=" ")
                        
        print("")
   
#Driver Code 
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern15(n)
    
if __name__ == "__main__":
    main()
           